resources/turso/README.md

```diff
@@ -0,0 +1,31 @@
+# Shuttle Turso
+
+This plugin allows services to connect to a Turso database. Turso is an edge-hosted distributed database based on libSQL.
+
+## Usage
+
+**IMPORTANT**: Currently Shuttle isn't able to provision a database for you (yet). This means you will have to create an account on their [website](https://turso.tech/) and follow the few steps required to create a database and create a token to access it.
+
+Add `shuttle-turso` to the dependencies for your service.
+This resource will be provided by adding the `shuttle_turso::Turso` attribute to your Shuttle `main` decorated function.
+
+It returns a `libsql_client::Client`. When running in production, the token used to connect to your database will be read from `TURSO_DB_TOKEN` in your `Secrets.toml` file. This can be overridden with the `token_secret` parameter. When running locally it will instantiate a local SQLite database of the name of your service instead of connecting to your edge database.
+
+If you want to connect to a remote database when running locally, you can specify the `local_addr` parameter. In that case, the token will be read from your `Secrets.dev.toml` file.
+
+### Example
+
+```rust
+use libsql_client::client::Client;
+
+#[shuttle_runtime::main]
+async fn app(#[shuttle_turso::Turso(addr="libsql://advanced-lightspeed.turso.io")] client: Client) -> __ { }
```

Comment:
We can use secrets string interpolation here, see [this section](https://docs.shuttle.rs/resources/shuttle-aws-rds#parameters) in the rds docs for an example. I think we should also add the token param to the example and interpolate it from secrets, and maybe also make that one required.

I'm also thinking we should use one of the frameworks for the example so it's clear that using turso resource doesn't change the return type, what do you think?
```suggestion
async fn app(#[shuttle_turso::Turso(addr="libsql://advanced-lightspeed.turso.io")] client: Client) -> ShuttleAxum { ... }
```

Reply:
Perhaps we don't need to use secrets interpolation for the address, since it doesn't contain secrets like a sql connection string, rather just the token

Reply:
Yes, Pieter implemented it a while ago but we haven't really documented it very clearly. :smile: It's pretty cool!